Updating flags and docs for training on total energy targets

wood-b opened the pull request:
- Removes the `total_energy` flag from all OC22 base, base_joint, and example configs and documentation
- Uses the `oc20_ref` flag for logic in `OC22LmdbDataset` instead of `total_energy`    

Resolves #454

wood-b commented:
Don't merge yet, there was an issue with joint OC20 and OC22 training I want to check

mshuaibii reviewed (approved):
LGTM! 

Tested on my end with the public datasets.

wood-b commented:
> Don't merge yet, there was an issue with joint OC20 and OC22 training I want to check

Thanks @mshuaibii for double checking — we are good to merge on my end.

abhshkdz commented on TRAIN.md:
```diff
@@ -263,23 +263,21 @@ You can find examples configuration files in [`configs/oc22/s2ef`](https://githu
 ## Joint Training
 
-Training on OC20 total energies whether independently or jointly with OC22 requires `total_energy: True` and a path to the `oc20_ref` (download link provided below) to be specified in the configuration file. These are necessary to convert OC20 adsorption energies into their corresponding total energies. The following changes in the configuration file capture these changes:
+Training on OC20 total energies whether independently or jointly with OC22 requires a path to the `oc20_ref` (download link provided below) to be specified in the configuration file. These are necessary to convert OC20 adsorption energies into their corresponding total energies. The following changes in the configuration file capture these changes:
```
I'd keep the `total_energy: True|False` flag. It's much more explicit in conveying whether to train on total energies or not.

wood-b replied:
I added a `train_on_oc20_total_energies` flag.

abhshkdz commented on TRAIN.md:
```diff
@@ -263,23 +263,21 @@ You can find examples configuration files in [`configs/oc22/s2ef`](https://githu
     normalize_labels: False
-    total_energy: True
     #download at https://dl.fbaipublicfiles.com/opencatalystproject/data/oc22/oc20_ref.pkl
     oc20_ref: path/to/oc22_ref.pkl
```
`path/to/oc20_ref.pkl`?

wood-b replied:
I updated `path/to/oc22_ref.pkl` to be `path/to/oc20_ref.pkl`

abhshkdz commented on configs/is2re/example.yml:
```diff
@@ -38,7 +38,6 @@ dataset:
     # energies `oc20_ref` must be specificed to unreference existing OC20 data.
     # OC22 by default trains on total energy, so this flag is not necessary.
-    total_energy: False                                                         # True or False
     oc20_ref: False                                                             # True or False
```
`oc20_ref` is a string and not True / False right?

wood-b replied:
I updated this to `path to oc20_ref`

abhshkdz commented on ocpmodels/datasets/oc22_lmdb_dataset.py:
```diff
@@ -155,15 +154,15 @@ def __getitem__(self, idx):
         if attr == "y":
             # OC20 data
-            if "oc22" not in data_object and self.train_total:
+            if "oc22" not in data_object and self.oc20_ref:
```
Same comment as above. It seems like a non-obvious assumption that someone wants to train on total energies if they have the `oc20_ref` flag specified. `total_energy: True|False` is much more explicit.

wood-b commented:
Updated changes:

- Removes the `total_energy` flag in favor of `train_on_oc20_total_energies` 
- Updated documentation around training models with total energy targets

wood-b commented:
I made a few minor changes to docs and added some asserts for more descriptive errors based on testing. I tested on OC20+OC22 Joint, OC20 Total Energy, and OC22 Total Energy (default). 

I added two asserts to cover these scenarios:

- When training an OC20+OC22 Joint model (IS2RE or S2EF) and the `train_on_oc20_total_energies` does not get set to `True`. The error message previously was related to the data object missing a certain attribute.
- When training an OC20 total energy model and `train_on_oc20_total_energies` and `oc20_ref` are set correctly, but the dataset was set to `lmdb` or `trajectory_lmdb`, in this case the model would actually train on adsorption energy targets, now it throws an error.

abhshkdz reviewed (approved):
Looks great!
